src/structures/ThreadChannel.js

```diff
@@ -62,35 +63,35 @@ class ThreadChannel extends Channel {
     if ('parent_id' in data) {
       /**
        * The id of the parent channel of this thread
-       * @type {Snowflake}
+       * @type {?Snowflake}
        */
       this.parentId = data.parent_id;
     }
 
     if ('thread_metadata' in data) {
       /**
        * Whether the thread is locked
-       * @type {boolean}
+       * @type {?boolean}
```

Comment:
This can't be undefined

Reply:
It only is when received from an interaction because thread_metadata is not present

Reply:
You `??` it tho

Reply:
correct, because it's possible it doesn't exist within thread_metadata.

Reply:
If thread_metadata is not present, wouldn’t that mean that accessing `locked` in an undefined property would result in an error? (Same for all the other properties that come from thread_metadata)

Reply:
Literally 3 lines above where this suggestion it checks this.

Reply:
Oops my bad. Then where are these set to null?

Reply:
It isn't, the types are correct. For docs the ? means nullish, either null or undefined.

Reply:
Alright but usually the library sets missing properties to null at the top of the file so idk why that isn’t being done here but both work tbh

Reply:
Usually we set them to null if this.\<prop> isn't defined in the payload we received and has never been defined on the class already